service/cloudconfig/service.go

```diff
@@ -100,20 +100,37 @@ func (c CloudConfig) NewMasterCloudConfig(customObject providerv1alpha1.AzureCon
 		Hyperkube: k8scloudconfig.Hyperkube{
 			Apiserver: k8scloudconfig.HyperkubeApiserver{
 				BindAddress: "0.0.0.0",
-				Docker: k8scloudconfig.HyperkubeDocker{
-					RunExtraArgs: []string{
-						"-v /etc/kubernetes/config:/etc/kubernetes/config",
-						"-v /var/lib/waagent/ManagedIdentity-Settings:/var/lib/waagent/ManagedIdentity-Settings:ro",
+				Pod: k8scloudconfig.HyperkubePod{
+					HostExtraMounts: []k8scloudconfig.HostMount{
+						k8scloudconfig.HostMount{
+							Name:     "k8s-config",
+							Path:     "/etc/kubernetes/config/",
+							ReadOnly: true,
+						},
+						k8scloudconfig.HostMount{
+							Name:     "identity-settings",
+							Path:     "/var/lib/waagent/",
+							ReadOnly: true,
+						},
 					},
 					CommandExtraArgs: []string{
 						"--cloud-config=/etc/kubernetes/config/azure.yaml",
 					},
 				},
 			},
 			ControllerManager: k8scloudconfig.HyperkubeControllerManager{
-				Docker: k8scloudconfig.HyperkubeDocker{
-					RunExtraArgs: []string{
-						"-v /var/lib/waagent/ManagedIdentity-Settings:/var/lib/waagent/ManagedIdentity-Settings:ro",
+				Pod: k8scloudconfig.HyperkubePod{
+					HostExtraMounts: []k8scloudconfig.HostMount{
+						k8scloudconfig.HostMount{
+							Name:     "k8s-config",
+							Path:     "/etc/kubernetes/config/",
+							ReadOnly: true,
+						},
+						k8scloudconfig.HostMount{
+							Name:     "identity-settings",
+							Path:     "/var/lib/waagent/",
+							ReadOnly: true,
+						},
 					},
 					CommandExtraArgs: []string{
 						"--cloud-config=/etc/kubernetes/config/azure.yaml",
```

Comment:
We are using `azure.yaml` here, but don't have a mount. Is that correct?

Reply:
it is in k8scloudconfig already.

https://github.com/giantswarm/k8scloudconfig/blob/master/v_3_0_0/master_template.go#L2210